### PR TITLE
fuzzy testing

### PR DESCRIPTION
**Pull request by darxriggs**
JUST FOR DISCUSSION.

Currently this cannot be used out of the box due to some side effects in packetbeat (channels & buffers).

I had to patch some files:
* comment out code in https://github.com/elastic/libbeat/blob/master/common/cache.go#L212-L228
* put a `return` in here https://github.com/elastic/packetbeat/blob/master/protos/thrift/thrift.go#L1137

### CODE REVIEW

**Comment by ruflin** [comment]
@darxriggs I think this is definitively very interesting to get it in and also run it as part of CI. The changes in libbeat should probably be made configurable, so it can be invoked with the right flags.

File: Makefile [hunk 1]
```diff
@@ -75,6 +78,16 @@ coverage:
 benchmark:
 	$(GODEP) go test -short -bench=. ./... -cpu=2
 
+.PHONY: testfuzz
+testfuzz:
+	mkdir -p go-fuzz/workdir/corpus
+	$(GODEP) go build -o go-fuzz/workdir/prepare_corpus go-fuzz/prepare_corpus.go
```
Comment (urso) [inline Makefile]: master branche switched to glide + go vendor experiment

**Comment by urso** [comment]
does it really make sense to push traffic through complete chain for testing, or we rather want more localized tests on parsers (less size effects)?

**Comment by urso** [comment]
We're all for having fuzzy tests in packetbeat, as we see big benefits from testing protocol parsers via fuzzy testing.

But this proposal unfortunately adds too many moving parts having too many side effects. Instead we'd like to consider a more unit-test like approach to fuzzy testing on per protocol basis.

Thanks for opening this up and starting discussions!

**Pull request closed without merging**